spec/requests/api/v2/submissions_spec.rb

```diff
@@ -2,44 +2,385 @@
 
 require 'rails_helper'
 require './spec/requests/api/v2/shared_examples/api_key_authenticatable'
+require './spec/requests/api/v2/shared_examples/requests'
 
 describe 'Submissions API', with: :api_v2 do
-  let(:base_endpoint) { '/api/v2/submissions' }
+  let(:model_class) { Submission }
+  let(:base_endpoint) { "/api/v2/#{resource_type}" }
+  let(:resource_type) { model_class.name.demodulize.pluralize.underscore }
 
   it_behaves_like 'ApiKeyAuthenticatable'
 
-  context 'with multiple submissions' do
-    before { create_list(:submission, 5) }
+  context 'with a list of resources' do
+    let(:resource_count) { 5 }
+    let!(:resources) { create_list(:submission, resource_count) }
 
-    it 'sends a list of submissions' do
-      api_get base_endpoint
+    describe '#GET all resources' do
+      before { api_get base_endpoint }
 
-      # test for the 200 status-code
-      expect(response).to have_http_status(:success)
+      it 'responds with a success http code' do
+        expect(response).to have_http_status(:success)
+      end
 
-      # check to make sure the right amount of messages are returned
-      expect(json['data'].length).to eq(5)
+      it 'returns all the resources' do
+        expect(json['data'].count).to eq(resource_count)
+      end
     end
 
-    # Check filters, ESPECIALLY if they aren't simple attribute filters
+    describe '#filter' do
+      let(:target_resource) { resources.sample }
+
+      shared_examples 'it filters the resources correctly' do
```

Comment:
This shared example seems to declared in `spec/requests/api/v2/qc_files_spec.rb`, `spec/requests/api/v2/submissions_spec.rb` and in `spec/requests/api/v2/shared_behaviour/labware_shared_behaviour_spec.rb`. 

And all of them seem to be doing the following:

1. Checking if the response is an HTTP succes.
2. Checking if the call returns only one resource.
3. Checking if the call returns the correct resource.

Maybe we can move all these declarations into somewhere in `shared_behavior` directory and reuse it?

Reply:
In case @StephenHulme picks this up (please feel free), I've implemented something like this in my Qcable work.  So probably best to reuse that and go through existing tests, looking to replace this pattern with the shared example.

Shared example:
https://github.com/sanger/sequencescape/blob/Y24-190-support-limber-v2-qcable/spec/requests/api/v2/shared_examples/requests.rb#L138-L150

Usage:
https://github.com/sanger/sequencescape/blob/Y24-190-support-limber-v2-qcable/spec/requests/api/v2/qcables_spec.rb#L30-L57